crates/ra_hir/src/ty/method_resolution.rs

```diff
@@ -174,4 +174,25 @@ impl Ty {
         }
         None
     }
+
+    // This would be nicer if it just returned an iterator, but that's really
+    // complicated with all the cancelable operations
```

Comment:
That comment is partially obsolete: we don't have result-based cancelation any more, but we still  run into lifetimes problems, see #802